sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/model/sql/adapter/ModelFieldsAdapters.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.nexmark.model.sql.adapter;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.beam.sdk.nexmark.model.Auction;
+import org.apache.beam.sdk.nexmark.model.Bid;
+import org.apache.beam.sdk.nexmark.model.Person;
+
+/**
+ * Maps Java model classes to Beam SQL record types.
+ */
+public class ModelFieldsAdapters {
+
+  public static final Map<Class, ModelFieldsAdapter> ADAPTERS =
+      ImmutableMap.<Class, ModelFieldsAdapter>builder()
+          .put(Auction.class, auctionAdapter())
+          .put(Bid.class, bidAdapter())
+          .put(Person.class, personAdapter())
+          .build();
+
+  private static ModelFieldsAdapter<Person> personAdapter() {
+    return new ModelFieldsAdapter<Person>(
+        BeamRecordSqlTypeBuilder.of()
+            .withLongField("id")
+            .withStringField("name")
```

Comment:
<!--thread_id:cc_151126910_t; commit:050bdab2afe544b8a5c58bfe1dfe1914a9820c8f; resolved:1-->
<!--section:context-quote-->
> **echauchot** wrote:
> You're right it is out of the scope of this PR. all my point was to make the fields mapping generic enough to be included in sql package (like you extracted the other feature) rather than coding it for each pojo. Like you said above implementing `getFieldValue(pojo, fieldName)` using reflection might be a good thing if it is feasible.

<!--section:body-->
This discussion made me start implementation of [automatic BeamRecordType generation](https://github.com/apache/beam/pull/4204). Keeping these classes as they are right now until record type generation is ready.

Reply:
+1